lib/SILOptimizer/Mandatory/TFLowerGraph.cpp

```diff
@@ -1881,9 +1876,58 @@ GLStatus TFGraphLowering::visitGraphOperationInst(GraphOperationInst *inst) {
         break;
       }
       case SymbolicValue::Array: {
-        // FIXME: Handle array attributes.
+        CanType elementType;
+        auto rawElements = attrValue.getArrayValue(elementType);
+        SmallVector<SymbolicValue, 4> elements;
+        elements.reserve(rawElements.size());
+        for (auto elt : rawElements)
+          elements.push_back(elt.lookThroughSingleElementAggregates());
+
+        auto elementTypeString = elementType->getString();
+
+        // TODO: TF_SetAttrTypeList
+
+        if (elementTypeString == "String") {
```

Comment:
is there any enum that we can match on, rather than doing (sub)string comparison?

Reply:
Unfortunately, not really (that I know of).  This really comes down to what types we want to allow in #tfop attribute lists.  We could tighten this up to check that the types are coming from Swift module.  I'll dig around to see what ASTContext has to work with here.